lib/ash_json_api/controllers/helpers.ex

```diff
@@ -425,13 +425,13 @@ defmodule AshJsonApi.Controllers.Helpers do
 
           query =
             query
-            |> Ash.Query.load(fields_to_load)
-            |> Ash.Query.set_context(request.context)
             |> Ash.Query.for_read(
               action,
               Map.merge(request.arguments, params),
               Keyword.put(Request.opts(request), :page, false)
             )
+            |> Ash.Query.load(fields_to_load)
+            |> Ash.Query.set_context(request.context)
```

Comment:
I'm not sure if this is a bug in the ash core, or if this was an issue with how `for_read` was being used here. If `Ash.Query.load` with the calculation args came _before_ the `for_read`, `for_read` would smash the calculation args in some scenarios 🤔 Doing `for_read` before the `load` call seems to resolve it - I assume there's some nuance to how Ash handles queries here that's eluding me 😅

Reply:
oh, interesting. 🤔 that sounds like a bug then, unfortunately. For now, you can put the `load` after the `for_read` call, but we the `set_context` does need to go before it.

Reply:
@zachdaniel sounds good, I just updated that!